code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -144,22 +149,26 @@
 	else
 		. = oview(vision_range, target_from)
 
-/mob/living/simple_animal/hostile/proc/FindTarget(list/possible_targets, HasTargetsList = 0)//Step 2, filter down possible targets to things we actually care about
-	. = list()
-	if(!HasTargetsList)
+/mob/living/simple_animal/hostile/proc/FindTarget(list/possible_targets)//Step 2, filter down possible targets to things we actually care about
+	var/list/all_potential_targets = list()
+
+	if(isnull(possible_targets))
 		possible_targets = ListTargets()
-	for(var/pos_targ in possible_targets)
-		var/atom/A = pos_targ
-		if(Found(A))//Just in case people want to override targetting
-			. = list(A)
+
+	for(var/atom/target_sources as anything in possible_targets)
```

Comment:
Why is this separate loop, nothing uses possible targets later

Reply:
The for loop right under it does, but I still should merge them